payments-core/src/test/java/com/stripe/android/networking/StripeApiRepositoryTest.kt

```diff
@@ -2542,6 +2542,78 @@ internal class StripeApiRepositoryTest {
             }
         }
 
+    @Test
+    fun `Verify customer session client secret not in params when null`() = runTest {
+        val stripeResponse = StripeResponse(
+            200,
+            ElementsSessionFixtures.EXPANDED_PAYMENT_INTENT_JSON.toString(),
+            emptyMap()
+        )
+
+        whenever(stripeNetworkClient.executeRequest(any<ApiRequest>())).thenReturn(stripeResponse)
+
+        create().retrieveElementsSession(
+            params = ElementsSessionParams.PaymentIntentType(
+                clientSecret = "client_secret",
+                customerSessionClientSecret = null,
+                externalPaymentMethods = null,
+            ),
+            options = DEFAULT_OPTIONS,
+        )
+
+        verify(stripeNetworkClient).executeRequest(apiRequestArgumentCaptor.capture())
+
+        val request = apiRequestArgumentCaptor.firstValue
+        val params = requireNotNull(request.params)
+
+        assertEquals(
+            "https://api.stripe.com/v1/elements/sessions",
+            request.baseUrl
+        )
+
+        with(params) {
+            assertThat("payment_intent").isEqualTo(this["type"])
```

Comment:
These asserts are backwards. Mind cleaning them up in a follow up? I think it's true for the whole file.

Reply:
Oof yeah. I was wondering why these tests were felt odd. Added a [task](https://jira.corp.stripe.com/browse/MOBILESDK-1984).